receiver/awsfirehosereceiver/internal/unmarshaler/cwlog/unmarshaler.go

```
@@ -70,22 +70,40 @@ func (u *Unmarshaler) UnmarshalLogs(compressedRecord []byte) (plog.Logs, error)
 	byResource := make(map[resourceKey]plog.LogRecordSlice)
 
 	// Multiple logs in each record separated by newline character
+	var anyErrors bool
```

Naive question, would it be possible, safe and desirable to use a multi error here instead? I know it has specific logs per error type, but the returned value is still the generic `errInvalidRecords`, and I'm wondering if a more specific error would be somehow helpful for the API responses.

---

IMO the receiver should not be logging and swallowing these errors, it should return them to the client. Such errors most likely imply that either the collector has been misconfigured with the wrong encoding, or the client has been misconfigured to send the wrong type of data.

While it's possible to determine the causes of these by looking at the collector logs, that's not necessarily practical. The person who configures things on the client side (i.e. CloudWatch/Firehose) may not be the same person who is running the collector, and may not have access to the logs. In this case it would be more helpful to respond with an error to the client, so they can understand what they've done wrong.

All that to say: yes, I think we should return more specific errors, but that would be a significant change in behaviour and we should do it in a different breaking changes PR.